apps/auth/oidc/auth.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from flask import g, request
+from flask_oidc_ex import OpenIDConnect
+
+import superdesk
+from apps.auth.errors import CredentialsAuthError
+from apps.auth.service import AuthService
+from eve.utils import config
+from superdesk import get_resource_service
+from superdesk.resource import Resource
+from superdesk.utils import ignorecase_query
+
+
+class OIDCAuthResource(Resource):
+    schema = {
+        'token': {
+            'type': 'string'
+        },
+        'user': Resource.rel('users', True)
+    }
+    resource_methods = ['POST']
+    public_methods = ['POST']
+    extra_response_fields = ['user', 'token', 'username']
+
+
+superdesk.intrinsic_privilege('auth_oidc', method=['DELETE'])
+
+
+class OIDCAuthService(AuthService):
+    def __init__(self, datasource=None, backend=None, app=None):
+        super().__init__(datasource=datasource, backend=backend)
+        self.oidc = OpenIDConnect(app)
+
+    def authenticate(self, credentials):
+        auth_header = request.headers.get('Authorization', '').split(' ', 1)
+        if auth_header[0] != 'Bearer' and len(auth_header) != 2:
+            raise CredentialsAuthError(credentials)
+        token = auth_header[1]
+        is_valid = self.oidc.validate_token(token, ['openid', 'email', 'profile'])
+        if not is_valid:
+            raise CredentialsAuthError(credentials)
+
+        users_service = get_resource_service('users')
+        username = g.oidc_token_info.get('username')
```

Comment:
is `username` required here?

Reply:
Yes, we need to check if user is exist and sync user data from `keycloak`, so we could use either `username` or `email` here to find user